pkg/api/types.go

```diff
@@ -1895,6 +1901,10 @@ type DeleteOptions struct {
 	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
 	// specified type will be used.
 	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
+
+	// Must be fulfilled before a deletion is carried out. If not possible, a 409 Conflict status will be
+	// returned.
+	Preconditions Preconditions `json:"preconditions"`
```

Comment:
Argh, yes, I should have noticed. At least the external one should be that
way.

On Mon, Mar 21, 2016 at 12:11 PM, Brian Grant <notifications@github.com>
wrote:

> In pkg/api/types.go
> <https://github.com/kubernetes/kubernetes/pull/22965#discussion_r56879314>
> :
>
> > @@ -1895,6 +1901,10 @@ type DeleteOptions struct {
> >  	// The value zero indicates delete immediately. If this value is nil, the default grace period for the
> >  	// specified type will be used.
> >  	GracePeriodSeconds *int64 `json:"gracePeriodSeconds"`
> > +
> > +	// Must be fulfilled before a deletion is carried out. If not possible, a 409 Conflict status will be
> > +	// returned.
> > +	Preconditions Preconditions `json:"preconditions"`
>
> Should be a pointer. And, as long as we have json tags, both Preconditions
> and GracePeriodSeconds should be tagged with omitempty.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/22965/files/301ac178e596509962bad4cc8be311df1c9ce8df#r56879314>
>